pageserver/src/walredo.rs

```diff
@@ -156,6 +179,9 @@ impl WalRedoManager for PostgresRedoManager {
         base_img: Option<Bytes>,
         records: Vec<WALRecord>,
     ) -> Result<Bytes, WalRedoError> {
+        let start_time;
+        let lock_time;
+        let end_time;
```

Comment:
What's the point of those "forward declarations"? I understand this is how we'd probably do this in ansi C, but IMO it goes against the principle of variable locality: we should bind variables as close to their use site as possible.

Reply:
Matter of style, I guess. The forward declaration is needed for 'lock_time', because that's assigned in the block below that's used as the scope of holding the lock on 'self.process'. With that, it seemed nicer to me to declare all the time-variables here together.